datastore/gcloud/aio/datastore/query.py

```diff
@@ -5,11 +5,60 @@
 from gcloud.aio.datastore.constants import MoreResultsType
 from gcloud.aio.datastore.constants import ResultType
 from gcloud.aio.datastore.entity import EntityResult
+from gcloud.aio.datastore.filter import Filter
 from gcloud.aio.datastore.utils import make_value
 from gcloud.aio.datastore.utils import parse_value
 
 
-class GQLQuery:
+class _BaseQuery:
+    json_key = ''
+
+    def __repr__(self) -> str:
+        return str(self.to_repr())
+
+    @classmethod
+    def from_repr(cls, data: Dict[str, Any]) -> '_BaseQuery':
+        raise NotImplementedError
+
+    def to_repr(self) -> Dict[str, Any]:
+        raise NotImplementedError
+
+
+# https://cloud.google.com/datastore/docs/reference/data/rest/v1/projects/runQuery#Query
+class Query(_BaseQuery):
+    json_key = 'query'
+
+    def __init__(self, kind: str, query_filter: Filter = None):
+        if not kind:
+            raise NotImplementedError('Kindless queries are not supported')
+        self.kind = kind
+        self.query_filter = query_filter
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Query):
+            return False
+        return bool(
+            self.kind == other.kind
+            and self.query_filter == other.query_filter)
+
+    @classmethod
+    def from_repr(cls, data: Dict[str, Any]) -> 'Query':
+        kind = data['kind']
+        query_filter = Filter.from_repr(data['filter'])
+        return cls(kind=kind, query_filter=query_filter)
+
+    def to_repr(self) -> Dict[str, Any]:
+        data = {
+            'kind': [{'name': self.kind}]
+        }
+        if self.query_filter:
+            data['filter'] = self.query_filter.to_repr()
+        return data
+
+
+class GQLQuery(_BaseQuery):
```

Comment:
Please add a comment point to `https://cloud.google.com/datastore/docs/reference/data/rest/v1/projects/runQuery#GqlQuery`, thanks! We must have missed it in the first pass.